### PR TITLE
Add support for `NoDefaultDate` option that excludes the sending of the `Date` header

### PR DESCRIPTION
**Pull request by phuslu**
same as https://github.com/valyala/fasthttp/pull/629  but with tests.

### CODE REVIEW

**Comment by phuslu** [comment]
For changes in https://github.com/valyala/fasthttp/pull/758/files#diff-34c6b408d72845d076d47126c29948d1R919-R928 , just make GoSec happy.

File: server.go [hunk 1]
```diff
@@ -907,9 +914,18 @@ func SaveMultipartFile(fh *multipart.FileHeader, path string) error {
...
 	_, err = copyZeroAlloc(ff, f)
-	return err
+	if err != nil {
+		return err
+	}
+
+	err = ff.Close()
+	if err != nil {
+		return err
+	}
```
Comment (erikdubbelboer) [inline server.go]: Now `ff.Close()` isn't called if `copyZeroAlloc` fails which could cause files to stay open (multipart files are backed by os files). I'm not sure why this wasn't an issue before. Doesn't changing it to `defer ff.Close() //nolint:errcheck` fix the issue?

Comment (phuslu) [reply server.go]: Done

**Comment by phuslu** [comment]
Hi @erikdubbelboer, I amended the commit. please have a look.

**Comment by erikdubbelboer** [comment]
Thanks!

**Comment by phuslu** [comment]
With this PR, we could minimize fasthttp response by following codes
```golang
server := &fasthttp.Server{
	NoDefaultServerHeader: true,
	NoDefaultContentType:  true,
	NoDefaultDate:         true,
	DisableKeepalive:      false,
}

// ... ...

ctx.SetStatusCode(fasthttp.StatusNoContent)
```
will give output:
```
HTTP/1.1 204 No Content

```

**Pull request merged**